shared/app/tab-bar/index.render.js.flow

```diff
@@ -1,12 +1,13 @@
 // @flow
 import React, {Component} from 'react'
 import type {Tab} from '../../constants/tabs'
+import type {Map} from 'immutable'
 
 export type Props = {
   onTabClick: (tab: Tab) => void,
   selectedTab: Tab,
   username?: string,
-  badgeNumbers: {[key: Tab]: number},
+  badgeNumbers: Map<Tab, number>,
```

Comment:
That's what the selector is for.

Having a selector like:

```js
const mapStateToProps = createSelector(
 [state => state.badgeNumbers,
  usernameSelector // from constants/selectors
  ],
  (badgeNumbers, username) => ({badgeNumbers: badgeNumbers.toObject(), username})
)
```

means the component won't receive a new object if the inputs haven't changed.

(https://github.com/reactjs/reselect#createselectorinputselectors--inputselectors-resultfunc)